File: Projeto_2/src/seguro.js
class Seguro{
    
    constructor(carro, cliente, valor, vigencia){
        this.carro = carro
        this.cliente = cliente
        this.valor = valor
        this.vigencia = vigencia
    }

    calcularValor(baseValor, taxa) {
        return baseValor + (baseValor * taxa);
      }
    
      verificarVigencia() {
        const hoje = new Date();
        const [inicio, fim] = this.vigencia.split(" a ").map(data => new Date(data));
        return hoje >= inicio && hoje <= fim;
      }
}

export default Seguro;

File: seguro.py
class Seguro:
    def __init__(self, carro, cliente, valor, vigencia):
        self.carro = carro
        self.cliente = cliente
        self.valor = valor
        self.vigencia = vigencia

    def calcular_valor(self, base_valor, taxa):
        return base_valor + (base_valor * taxa)

    def verificar_vigencia(self):
        return "Seguro válido"
